test/ArrayField_test.js

```diff
@@ -295,6 +295,41 @@ describe("ArrayField", () => {
       expect(inputs[2].id).eql("root_foo_1_bar");
       expect(inputs[3].id).eql("root_foo_1_baz");
     });
+
+    it("should render enough inputs with proper defaults to match minItems in schema when no formData is set", () => {
+      const complexSchema = {
+        type: "object",
+        definitions: {
+          Thing: {
+            type: "object",
+            properties: {
+              name: {
+                type: "string",
+                default: "Default name"
+              }
+            }
+          }
+        },
+        properties: {
+          foo: {
+            type: "array",
+            minItems: 2,
+            items: {
+              $ref: "#/definitions/Thing"
+            }
+          }
+        }
+      };
+      let form = createFormComponent({schema: complexSchema, formData: { }});
+      let inputs = form.node.querySelectorAll("input[type=text]");
+      expect(inputs[0].value).eql("Default name");
+      expect(inputs[1].value).eql("Default name");
+
+      // whenever a value is set, honor it!
+      form = createFormComponent({schema: complexSchema, formData: {foo: []}});
+      inputs = form.node.querySelectorAll("input[type=text]");
+      expect(inputs.length).eql(0);
```

Comment:
It would be better to distinguish between the two sets of assertions here, so we're better informed when one of these fails.

This can usually be achieved by either mounting a single component per `it` test.